tests/common.py

```diff
@@ -43,22 +56,39 @@ def pytest_synth_params(metafunc):
 
         metafunc.parametrize('synth', targets)
 
-def run_synth(synth, top=None, cwd='build', src_files=None, hdr_files=None):
+def run_synth(synth, top=None, cwd='build', src_files=None, hdr_files=None, defines=None,
+              real_type=RealType.FixedPoint):
     # set defaults
     if src_files is None:
         src_files = []
     if hdr_files is None:
         hdr_files = []
+    if defines is None:
+        defines = {}
+
+    # add to source files
+    if real_type == RealType.HardFloat:
```

Comment:
we would probably want to do something similar in anasymod at some point, so that the HardFloat sources and headers are automatically pulled in if needed (but only if necessary, because we don't want HardFloat to be required)

Reply:
Sure makes sense. I think this can be easily managed, if we expose hard_float as an argument/option similarly to --float.